Sources/MapboxMaps/Style/Style.swift

```diff
@@ -40,14 +40,11 @@ public class Style {
                 success case. Else, returns a `LayerError` in the `Result` failure case.
      */
     @discardableResult
-    public func addLayer<T: Layer>(layer: T, layerPosition: LayerPosition? = nil) -> Result<Bool, LayerError> {
-
+    public func addLayer(layer: Layer, layerPosition: LayerPosition? = nil) -> Result<Bool, LayerError> {
```

Comment:
I want to make these methods throw instead of returning `Result`. Not for this PR though.

Only downside is the errors would be untyped, but I'm not convinced that's a big deal. It would clean up the error handling significantly.

Reply:
Ticketed.